src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/services/citizen/SettlementReachedCitizenNotificationServiceTest.java

```diff
@@ -0,0 +1,162 @@
+import uk.gov.hmcts.cmc.claimstore.events.offer.AgreementCountersignedEvent;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.offers.MadeBy;
+import uk.gov.hmcts.cmc.domain.models.offers.Settlement;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaimData;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleResponse;
+import uk.gov.hmcts.cmc.domain.models.sampledata.offers.SampleOffer;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class SettlementReachedCitizenNotificationServiceTest extends MockSpringTest {
+
+    private static final byte[] PDF_CONTENT = {1, 2, 3, 4};
+
+    @Autowired
+    private AgreementCounterSignedCitizenActionsHandler handler;
+
+    @Captor
+    private ArgumentCaptor<String> senderArgument;
+
+    @Captor
+    private ArgumentCaptor<String> templateArgument;
+
+    @Captor
+    private ArgumentCaptor<String> referenceArgument;
+
+    @Captor
+    private ArgumentCaptor<Map<String, String>> emailDataArgument;
+
+    private Claim claim;
+    private AgreementCountersignedEvent event;
+
+    @Before
+    public void setUp() {
+        Settlement settlement = new Settlement();
+        settlement.makeOffer(SampleOffer.validDefaults(), MadeBy.DEFENDANT);
+        settlement.accept(MadeBy.CLAIMANT);
+        settlement.countersign(MadeBy.DEFENDANT);
+
+        claim = SampleClaim
+            .builder()
+            .withClaimData(SampleClaimData.submittedByClaimant())
+            .withSettlementReachedAt(LocalDateTime.now())
+            .withResponse(SampleResponse.validDefaults())
+            .withSettlement(settlement)
+            .build();
+
+
+        when(pdfServiceClient.generateFromHtml(any(byte[].class), anyMap()))
+            .thenReturn(PDF_CONTENT);
+    }
+
+    @Test
+    public void shouldSendEmailToDefendantWithExpectedTemplateAndReferenceWhenCounterSignedByDefendant()
+        throws NotificationClientException {
+        event = new AgreementCountersignedEvent(claim, MadeBy.DEFENDANT);
+        handler.sendNotificationToOfferOriginator(event);
+
+        verify(notificationClient).sendEmail(templateArgument.capture(), senderArgument.capture(),
+            emailDataArgument.capture(), referenceArgument.capture());
+
+        assertThat(templateArgument.getValue()).isEqualTo("9d1ddac9-d6a7-41f3-bfd4-dcfbcb61dcf1");
+
+        assertThat(referenceArgument.getValue())
+            .isEqualTo("to-claimant-agreement-counter-signed-by-defendant-notification-000CM001");
```

Comment:
This is wrong, test says `shouldSendEmailToDefendantWithExpectedTemplateAndReferenceWhenCounterSignedByDefendant` while notification reference states notifications goes to claimant, while it should be defendant.

We should have 4 different combinations tested:
- `to-claimant-agreement-counter-signed-by-defendant-notification-000CM001` (notification for claimant)
- `to-defendant-agreement-counter-signed-by-defendant-notification-000CM001` (confirmation for defendant)
- `to-defendant-agreement-counter-signed-by-claimant-notification-000CM001` (notification for defendant - not supported ATM)
- `to-claimant-agreement-counter-signed-by-claimant-notification-000CM001` (confirmation for claimant - not supported ATM)